File: java/1694_ReformatPhoneNumber.java
class Solution {
    public String reformatNumber(String number) {
        String s = number.replaceAll("[^0-9]", "");
        StringBuilder sb = new StringBuilder(s);
        if (s.length() == 2 || s.length() == 3) return s;
        if (s.length() == 4) return s.substring(0,2) + "-" + s.substring(2);
        int n = s.length() / 3;
        for (int i = 1; i < n; i++) sb.insert(4*i-1,'-');
        if (sb.lastIndexOf("-") == sb.length() - 5 || sb.lastIndexOf("-") == sb.length() - 6) 
            sb.insert(sb.length()-2,'-');
        return sb.toString();
    }
}

File: python/1694_ReformatPhoneNumber.py
class Solution:
    def reformatNumber(self, number):
        output = []
        number = number.replace("-", "").replace(" ", "")
        length = len(number)
        i = 0
        while i < length:
            if i+4 == length:
                output.append(number[i:i+2])
                i += 2
            elif i+3 <= length:
                output.append(number[i:i+3])
                i += 3
            else:
                output.append(number[i:i+2])
                i += 2
                
        return output[0] if len(output) < 1 else "-".join(output)
